dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/injection/WebServiceContextWrapper.java

```diff
@@ -40,10 +46,19 @@ public EndpointReference getEndpointReference(Element... referenceParameters) {
         return context.getEndpointReference(clazz, referenceParameters);
     }
 
-    /** {@inheritDoc} */
     @Override
     public MessageContext getMessageContext() {
-        return context.getMessageContext();
+
+        WrappedMessageContext wmc = (WrappedMessageContext) context.getMessageContext();
+
+        if (wmc != null) {
+            Message msg = wmc.getWrappedMessage();
+            SOAPMessageContextImpl smci = new SOAPMessageContextImpl(msg);
+            return smci;
+        } else {
+            return context.getMessageContext();
```

Comment:
```suggestion
            return wmc;
```

or just return null here.